### PR TITLE
[WIP] forward transactions to the mainnet

### PR DESCRIPTION
**Pull request by batrudinych**
Minimal example.js to verify `push-tx` endpoint:
```
'use strict'

const eos = require('eosjs')

const { port } = require('./config/moonbeam.conf.json')
const { getReq } = require('./test/helper')

const config = require('./config/dev-signing-ws.config.json')
const { getClient } = require('./dev-tools')

const req = getReq(port)

;(async () => {
  // mainchain verify and fwd tx
  const expireInSeconds = 35
  const user = 'testuser4321'
  const client = getClient(config)

  const txdata = {
    actions: [{
      account: 'testfaucet11',
      name: 'verify',
      authorization: [{
        actor: user,
        permission: 'active'
      }],
      data: {
        account: user
      }
    }]
  }

  const tx = await client.api.transact(txdata, {
    broadcast: false,
    blocksBehind: 3,
    expireSeconds: expireInSeconds
  })

  tx.serializedTransaction = eos.Serialize.arrayToHex(tx.serializedTransaction)

  const payload = {
    t: tx,
    v: 1,
    code: 'TEST12'
  }

  console.log(await req('POST', '/push-tx', payload))
})()
```

### CODE REVIEW

File: lib/routes-priv.js [hunk 1]
```diff
@@ -431,6 +431,50 @@ class RoutesPriv {
     })
   }
 
+  async onPrivatePushTxHttpRequest (req, res) {
+    const rpc = this.mainnetRpc
+
+    const payload = req.body
+    const _v = Joi.validate(payload, schemaMainTx)
```
Comment (vigan-abd) [inline lib/routes-priv.js]: As I see [schemaMainTx](https://github.com/bitfinexcom/moonbeam/pull/53/files#diff-aacdac560145b2e6e58f5d1fea0b0821R23) shouldn't have .single() method I think, based on [docs](https://github.com/hapijs/joi/blob/master/API.md#arraysingleenabled) I see that it accepts also non array values, but [push transaction](https://developers.eos.io/manuals/eos/latest/nodeos/plugins/chain_api_plugin/api-reference/index#operation/push_transaction) requires that `signatures` field to be array

File: lib/moonbeam.js [hunk 2]
```diff
@@ -120,6 +120,7 @@ class Moonbeam {
     app.post('/sm-tos', cmw, privRoutes.onPrivateTosMainSetHttpRequest.bind(privRoutes))
     app.post('/gm-tos', cmw, privRoutes.onPrivateTosMainGetHttpRequest.bind(privRoutes))
     app.post('/login', cmw, privRoutes.onPrivateLoginHttpRequest.bind(privRoutes))
+    app.post('/push-tx', cmw, privRoutes.onPrivatePushTxHttpRequest.bind(privRoutes))
```
Comment (vigan-abd) [inline lib/moonbeam.js]: Maybe we should add also integration test case for it?

File: lib/routes-priv.js (hunk 1 above)
Comment (batrudinych) [reply lib/routes-priv.js]: makes sense, will correct that

File: lib/moonbeam.js (hunk 2 above)
Comment (batrudinych) [reply lib/moonbeam.js]: agree here. However, this particular endpoint and the service in general will be reworked massively. Let's get this draft endpoint implementation merged first and implement the tests later on when we have at least some bits of logic inserted.

**Comment by batrudinych** [comment]
will be moved to `mainnet` branch

**Pull request closed without merging**